File: composeApp/src/iosMain/kotlin/com/coretechies/jaap/ui/MainScreenIOS.kt
import androidx.compose.ui.input.key.Key.Companion.Tab
import com.coretechies.jaap.screens.ListScreen // Ensure this is generated correctly

public struct MainScreenIos: View {
    @State private var selectedTab: Tab = .home // Use the dot notation for enum cases

    enum Tab {
        case home
                case list
                case menu
    }

    public var body: some View {
        TabView(selection: $selectedTab) {
        HomeScreen() // Make sure HomeScreen is correctly defined and accessible
            .tabItem {
                Label("Home", systemImage: "house")
            }
            .tag(Tab.home)

        // Use the generated Swift interface for ListScreen
        ListScreen() // Ensure this is the right usage for the generated Kotlin function
            .tabItem {
                Label("List", systemImage: "list.bullet")
            }
            .tag(Tab.list)

        MenuScreen() // Make sure MenuScreen is correctly defined and accessible
            .tabItem {
                Label("Menu", systemImage: "ellipsis.circle")
            }
            .tag(Tab.menu)
    }
        .accentColor(.blue) // Change color of selected tab icon
    }
}

File: composeApp/src/iosMain/kotlin/com/coretechies/jaap/ui/MainScreenIOS.swift
import SwiftUI

struct ContentView: View {
    @State private var selectedTab: Tab = .home
    @ObservedObject var prefs: PreferencesViewModel // Assuming you have a view model for prefs

    var body: some View {
        TabView(selection: $selectedTab) {
        HomeView(prefs: prefs) // Replace with your actual Home view
            .tabItem {
                Image("home") // Use your image assets
                Text("Home")
            }
            .tag(Tab.home)

        ListView(prefs: prefs) // Replace with your actual List view
            .tabItem {
                Image("list") // Use your image assets
                Text("List")
            }
            .tag(Tab.list)

        MenuView(prefs: prefs) // Replace with your actual Menu view
            .tabItem {
                Image("menu") // Use your image assets
                Text("Menu")
            }
            .tag(Tab.menu)
    }
        .accentColor(Color.orange) // Set the color for the selected tab
    }
}

enum Tab {
    case home
    case list
    case menu
}
